### PR TITLE
[W4.5c][W14-A1]Mukesh Gadupudi

### PR DESCRIPTION
**Pull request by shanmu9898**
Added code to handle exceptions. Some code adapted from #2.

### CODE REVIEW

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -111,7 +112,10 @@ private CommandResult executeCommand(Command command)  {
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        }catch (StorageWriteException swe) {
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java:115]: In general, it is good to handle the exception closer to where it occurs.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -111,7 +112,10 @@ private CommandResult executeCommand(Command command)  {
...
+            return new CommandResult(MESSAGE_STORAGE_ERROR);
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java:116]: In general, just showing the error message or printing the stack trace is not considered "good exception handling".
Think how you can improve the user experience in light of the exception that just occurred.

File: src/seedu/addressbook/common/Messages.java [hunk 3]
```diff
@@ -15,4 +15,6 @@
             "java seedu.addressbook.Main [STORAGE_FILE_PATH]";
     public static final String MESSAGE_WELCOME = "Welcome to your Address Book!";
     public static final String MESSAGE_USING_STORAGE_FILE = "Using storage file : %1$s";
+    public static final String MESSAGE_STORAGE_ERROR = "Error while saving the file (Ensure the file is not given read only access";
```
Comment (okkhoy) [inline src/seedu/addressbook/common/Messages.java:18]: Good to see you have moved the message to the separate class, where it belongs. :+1:

File: src/seedu/addressbook/storage/StorageFile.java [hunk 4]
```diff
@@ -51,6 +51,15 @@ public StorageOperationException(String message) {
             super(message);
         }
     }
+    /**
+          * Signals that some error has occured while trying to write data between the application
```
Comment (okkhoy) [inline src/seedu/addressbook/storage/StorageFile.java:55]: While this is a good header comment, you can try to be concise.

**Comment by okkhoy** [comment]
@shanmu9898 
Some comments added. Please ack & close the PR after reading comments.

**Comment by shanmu9898** [comment]
Yeap! Noted! Thank you sir.

**Pull request closed without merging**